dev/io.openliberty.security.jakartasec.4.0.internal/src/io/openliberty/security/jakartasec40/Utils.java

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.security.jakartasec40;
+
+import java.lang.annotation.Annotation;
+import java.util.Map;
+
+import com.ibm.websphere.ras.annotation.Sensitive;
+import com.ibm.ws.security.javaeesec.JavaEESecConstants;
+
+import jakarta.security.enterprise.identitystore.IdentityStore.ValidationType;
+import jakarta.security.enterprise.identitystore.InMemoryIdentityStoreDefinition;
+
+/**
+ * A class to assist with generating actual instances of an @InMemoryIdentityStoreDefinition,
+ * given a simple map of name (String) to value (Object) values.
+ *
+ * Two separate static methods are created, one to allow the return of a @Credentials structure,
+ * and the second to create the in memory identity store structure.
+ *
+ *
+ */
+public class Utils {
+
+    public static final String CALLER_NAME_NAME = "callerName";
+    public static final String GROUPS_NAME = "groups";
+    public static final String PASSWORD_NAME = "password";
+
+    public static InMemoryIdentityStoreDefinition.Credentials getInstanceOfCredentialsAnnotation(final Map<String, Object> overrides) {
+        InMemoryIdentityStoreDefinition.Credentials annotation = new InMemoryIdentityStoreDefinition.Credentials() {
+
+            @Override
+            public Class<? extends Annotation> annotationType() {
+                return null;
+            }
+
+            @Override
+            public String callerName() {
+                return (overrides != null &&
+                        overrides.containsKey(CALLER_NAME_NAME)) ? (String) overrides.get(CALLER_NAME_NAME) : "";
```

Comment:
```suggestion
                return (overrides == null ? "" :
                        overrides.getOrDefault(CALLER_NAME_NAME, ""));
```

A map can have `null` values.  Do we need to handle that?  i.e. does null mean something or will it just cause problems if it is null?

Reply:
Hi, the `overrides` map has its keys and values currently mapped from an annotation only (@InMemoryIdentityStoreDefinition) elements, where all element methods (which have default values or required user-supplied values in the case of callerName and password), are invoked, a `Map` created from method-name/method-value and the static Utils.java method called.

So the checks I've put in are probably overly cautious as `overrides` (current usage) will always be not-null, and keys names will always exist.  That said, future use may call these methods directly in other ways (like the JUnits currently do), so the checks are valid.

But, your first suggestion is reasonable, more compact and more efficient, so I'll update all the use cases.

Done.